src/kubernetes_charm.py

```diff
@@ -82,10 +76,19 @@ def _upgrade(self) -> typing.Optional[kubernetes_upgrade.Upgrade]:
         except upgrade.PeerRelationNotReady:
             pass
 
+    @property
+    def _substrate(self) -> str:
+        return "k8s"
+
+    def is_exposed(self, relation=None) -> typing.Optional[bool]:
+        """No-op since this charm is exposed with node-port"""
```

Comment:
changed the signature of `is_exposed` since relation is not used, but the workload's `reconcile` relies on this property and thus needs to return `None` in k8s charms